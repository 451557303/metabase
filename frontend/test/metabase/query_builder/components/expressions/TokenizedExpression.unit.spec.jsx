import React from "react";
import renderer from "react-test-renderer";

import TokenizedExpression from "metabase/query_builder/components/expressions/TokenizedExpression";

const TEST_EXPRESSIONS = [
  "Count",
  "Sum(Subtotal)",
  'Sum("Foo Bar")',
  "1 + Sum(2 + Total)",
];

describe("TokenizedExpression", () => {
  for (const expression of TEST_EXPRESSIONS) {
    it(`should render: ${expression}`, () => {
      const tree = renderer
        .create(<TokenizedExpression source={expression} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  }
});
