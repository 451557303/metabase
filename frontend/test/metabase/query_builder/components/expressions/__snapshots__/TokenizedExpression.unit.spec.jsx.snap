// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenizedExpression should render 1 + Sum(2 + Total) 1`] = `
<span
  className="Expression-node Expression-group"
>
  <span
    className="Expression-node Expression-token"
  >
    1
  </span>
  <span
    className="Expression-node Expression-token"
  >
     
  </span>
  <span
    className="Expression-node Expression-token"
  >
    +
  </span>
  <span
    className="Expression-node Expression-token"
  >
     
  </span>
  <span
    className="Expression-node Expression-aggregation Expression-tokenized"
  >
    <span
      className="Expression-node Expression-aggregation-name"
    >
      Sum
    </span>
    <span
      className="Expression-node Expression-group"
    >
      <span
        className="Expression-node Expression-open-paren"
      >
        (
      </span>
      <span
        className="Expression-node Expression-token"
      >
        2
      </span>
      <span
        className="Expression-node Expression-token"
      >
         
      </span>
      <span
        className="Expression-node Expression-token"
      >
        +
      </span>
      <span
        className="Expression-node Expression-token"
      >
         
      </span>
      <span
        className="Expression-node Expression-field Expression-tokenized"
      >
        Total
      </span>
      <span
        className="Expression-node Expression-close-paren"
      >
        )
      </span>
    </span>
  </span>
</span>
`;

exports[`TokenizedExpression should render Count 1`] = `
<span
  className="Expression-node Expression-group"
>
  <span
    className="Expression-node Expression-metric Expression-tokenized"
  >
    Count
  </span>
</span>
`;

exports[`TokenizedExpression should render Sum("Foo Bar") 1`] = `
<span
  className="Expression-node Expression-group"
>
  <span
    className="Expression-node Expression-aggregation Expression-tokenized"
  >
    <span
      className="Expression-node Expression-aggregation-name"
    >
      Sum
    </span>
    <span
      className="Expression-node Expression-group"
    >
      <span
        className="Expression-node Expression-open-paren"
      >
        (
      </span>
      <span
        className="Expression-node Expression-string-literal Expression-tokenized"
      >
        <span
          className="Expression-node Expression-open-quote"
        >
          "
        </span>
        <span
          className="Expression-node Expression-field"
        >
          Foo Bar
        </span>
        <span
          className="Expression-node Expression-close-quote"
        >
          "
        </span>
      </span>
      <span
        className="Expression-node Expression-close-paren"
      >
        )
      </span>
    </span>
  </span>
</span>
`;

exports[`TokenizedExpression should render Sum(Subtotal) 1`] = `
<span
  className="Expression-node Expression-group"
>
  <span
    className="Expression-node Expression-aggregation Expression-tokenized"
  >
    <span
      className="Expression-node Expression-aggregation-name"
    >
      Sum
    </span>
    <span
      className="Expression-node Expression-group"
    >
      <span
        className="Expression-node Expression-open-paren"
      >
        (
      </span>
      <span
        className="Expression-node Expression-field Expression-tokenized"
      >
        Subtotal
      </span>
      <span
        className="Expression-node Expression-close-paren"
      >
        )
      </span>
    </span>
  </span>
</span>
`;
